### ISSUE TITLE
ld: warning: no platform load command found in sys-o.a assuming macOS

### ISSUE THREAD

**Issue by IanButterworth** [JuliaLang/julia#51562]
I assume this is minor given it correctly assumes macOS

```
Output ──────  22.320461 seconds
    LINK usr/lib/julia/sys.dylib
ld: warning: no platform load command found in '/Users/ian/Documents/GitHub/julia/usr/lib/julia/sys-o.a[2](text#0.o)', assuming: macOS
ld: warning: no platform load command found in '/Users/ian/Documents/GitHub/julia/usr/lib/julia/sys-o.a[3](text#1.o)', assuming: macOS
ld: warning: no platform load command found in '/Users/ian/Documents/GitHub/julia/usr/lib/julia/sys-o.a[4](text#2.o)', assuming: macOS
ld: warning: no platform load command found in '/Users/ian/Documents/GitHub/julia/usr/lib/julia/sys-o.a[5](metadata.o)', assuming: macOS
ld: warning: no platform load command found in '/Users/ian/Documents/GitHub/julia/usr/lib/julia/sys-o.a[6](sysimg.o)', assuming: macOS
    JULIA stdlib/Unicode.release.image
```

macOS Sonoma
```
julia> versioninfo()
Julia Version 1.11.0-DEV.593
Commit f2d1276be8 (2023-10-03 12:04 UTC)
Platform Info:
  OS: macOS (arm64-apple-darwin23.0.0)
  CPU: 10 × Apple M2 Pro
  WORD_SIZE: 64
  LLVM: libLLVM-15.0.7 (ORCJIT, apple-m1)
  Threads: 1 on 6 virtual cores
```

**Comment by gbaraldi** [comment]
https://projects.blender.org/blender/blender/pulls/110243/files it seems we need to pass a linker flag

**Comment by vtjnash** [comment]
Apple is about 2 years behind with publishing their usual documentation for this sort of thing (at https://github.com/apple-oss-distributions/ld64), so we may not learn how to fix this for a while

Someone just observed (17 minutes ago) on a forum elsewhere on the internet that this means our target triple is not correct (missing the OS version)
https://discourse.llvm.org/t/expressing-the-darwin-macos-target-build-command-in-llvm-ir/74198/2?u=vtjnash

Which causes this part of LLVM not to function correctly: https://reviews.llvm.org/D29044

**Comment by gbaraldi** [comment]
Can we add the version there?

**Comment by vtjnash** [comment]
Yeah, we are explicitly using a platform there whose last released version was 10.4. I think we need the string "darwin18" (or later) to fix this.
